tools/configure.sh: Fix order of evaluation of enforce option

ghn-certi opened the pull request:
## Summary
According to the usage documentation, [-e] option should "enforce
distclean if already configured". This behavior was broken due to the -e
option being evaluated only if the current defconfig is different from
the backup defconfig.

## Impact
The configure.sh script will finally enforce the distclean process regardless of the current state of the defconfig file.
No impact if the user does not rely on the [-e] option.

## Testing
Validated the 3 possible paths:
* ./tools/configure.sh with -e
* ./tools/configure.sh after a modification to the defconfig file
* ./tools/configure.sh without any modification to the defconfig file

xiaoxiang781216 commented:
Should we change the document like this instead?
"enforce distclean if configuration change"
Our build script always interact with the build system like this:
```
./configure.sh -e chip:config
./make -jx
```
but we don't want to do the full build if there isn't configuration change.

Ouss4 commented:
I was thinking about this too but wanted to introduce a new option, maybe -E and keep -e as is.

xiaoxiang781216 commented:
-E/-e is a good idea.

ghn-certi commented:
> -E/-e is a good idea.

I've updated the PR.
Created a new command [-E] for the enforcement behavior.
Changed the documentation of [-e] to consider the distclean operation only if configration changed.

xiaoxiang781216 commented:
LGTM.
